Svc/Deframer/docs/sdd.md

```diff
@@ -376,7 +376,7 @@ assign these numbers.
 **Topology 3: Buffers containing packet data:**
 
 <div>
-<img src="img/top/file.png" width=1000/>
+<img src="img/top/deframer.png" width=1000/>
```

Comment:
`deframer.png` is not a good choice here. All the diagrams in the Deframer SDD pertain to the deframer. This one pertains to file packets. If necessary we can call it `deframer-file.png`.

At a higher level, can someone explain to me what is going on here? What is "Autodocs"? Why is it apparently requiring that every file in the entire F Prime repository have a unique name, disregarding the path where it is located? This is a severe restriction and IMO not a very reasonable one. Can this restriction be relaxed? If not, how can we avoid the situation where we merge an update to the docs and everything seems fine, and someone later reports that "Autodocs" has failed because of global name collisions?

Reply:
I can only point to the code that does the flattening:
https://github.com/nasa/fprime/blob/1f7b07da365bc23e10df4184c35d83147b4f1a1e/docs/doxygen/generate_docs.bash#L102
It was introduced in #1425.

Reply:
Sorry, here's the entrypoint for this (it was reachable from the log, but there's no reason to force you to jump through that): https://github.com/nasa/fprime/blob/devel/.github/workflows/autodocs.yml

Reply:
To answer the other question, the reason nothing warned you as you developed your PR is that it's only running on pushes to the development branch: https://github.com/nasa/fprime/blob/1f7b07da365bc23e10df4184c35d83147b4f1a1e/.github/workflows/autodocs.yml#L2-L7

If it also ran on pull requests, you'd have noticed.

---

The reason I'm paying any attention to this is because I checked to make sure my PR merged cleanly without fallout and I was surprised to see that it didn't. So I chased back from the error message (which is one I don't particularly recognize – neither busybox nor macOS generate it, it appears to be a feature of gnu-cp).

Reply:
Thanks for that detailed explanation. In the near term, it looks like we should rename the files. Longer term, we should look into making the doc generation more robust.

Reply:
I'm happy to use the names you suggested. I have very limited relation to the project, so I just needed a placeholder to hang a discussion off.